apiserver/facades/agent/instancemutater/lxdprofilewatcher.go

```diff
@@ -440,18 +427,12 @@ func (w *machineLXDProfileWatcher) charmChange(chURL string) error {
 		if err != nil {
 			return errors.Trace(err)
 		}
-		charmID, err := w.applicationService.GetCharmID(context.TODO(), applicationcharm.GetCharmArgs{
+
+		lxdProfile, _, err := w.applicationService.GetCharmLXDProfile(context.TODO(), applicationcharm.CharmLocator{
```

Comment:
Whilst you're here, make this context from the method.

Reply:
Done.